### PR TITLE
Github workflow for building the vembrane dists and publishing them to PyPI

### PR DESCRIPTION
**Pull request by tedil**

### CODE REVIEW

File: .github/workflows/pypi.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - name: Install dependencies
+        run:  |
+          python -m pip install --upgrade pip
+          python -m pip install poetry black flake8
+          poetry install --no-interaction
```
Comment (mbargull) [inline .github/workflows/pypi.yml:21]:
```suggestion
          python -m pip install poetry
```
Should suffice, right?

File: .github/workflows/pypi.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  build-and-publish:
+    name: build and publish python dists to PyPI
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
```
Comment (mbargull) [inline .github/workflows/pypi.yml:16]: Since you build `wheel`s: Is it intentional to only build for 3.7 and not 3.8?

**Comment by mbargull** [comment]
BTW, https://github.com/vembrane/vembrane/blob/c859e3d3b1e980bdc08e37cff082f6ec18b1a161/.github/workflows/main.yml#L6 lists `master`, not `main`, i.e., currently only PRs run the other CI job.

File: .github/workflows/pypi.yml (hunk 2 above)
Comment (tedil) [reply .github/workflows/pypi.yml:16]: tbh I have no clue what I'm doing w.r.t. python packaging

File: .github/workflows/pypi.yml (hunk 1 above)
Comment (tedil) [reply .github/workflows/pypi.yml:21]: yes, indeed! copied that from the main.yml

**Comment by tedil** [comment]
> BTW,
> 
> https://github.com/vembrane/vembrane/blob/c859e3d3b1e980bdc08e37cff082f6ec18b1a161/.github/workflows/main.yml#L6
> 
> lists `master`, not `main`, i.e., currently only PRs run the other CI job.

I think github has an alias for main <-> master, but we should change that.

File: .github/workflows/pypi.yml (hunk 2 above)
Comment (mbargull) [reply .github/workflows/pypi.yml:16]: > I have no clue what I'm doing w.r.t. python packaging

I can't remember ever hearing anyone say the opposite about themselves! I'm pretty sure we're all in the same boat even if some tell differently XD

Comment (tedil) [reply .github/workflows/pypi.yml:16]: I think we can restrict poetry to build either sdist or wheel only via the `--format` option, if that helps?

Comment (tedil) [reply .github/workflows/pypi.yml:16]: Or do we have to include multiple python versions in this step?

Comment (mbargull) [reply .github/workflows/pypi.yml:16]: Nvm, sorry, this builds universal wheels and as such doesn't need multiple runs for different versions.

**Pull request merged**